File: android/app/src/main/java/com/fitnessrider/ui/components/HandPositionBadge.kt
package com.fitnessrider.ui.components

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Path
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.fitnessrider.model.HandPosition
import com.fitnessrider.theme.CardBorder
import com.fitnessrider.theme.TextPrimary
import com.fitnessrider.theme.TextSecondary
import com.fitnessrider.theme.TopBarGreenDark

@Composable
fun HandPositionBadge(
    position: HandPosition,
    modifier: Modifier = Modifier,
    isCompact: Boolean = false,
    showTitle: Boolean = true
) {
    Row(
        modifier = modifier
            .background(Color.White, RoundedCornerShape(8.dp))
            .border(1.dp, CardBorder, RoundedCornerShape(8.dp))
            .padding(
                horizontal = if (isCompact) 8.dp else 12.dp,
                vertical = if (isCompact) 4.dp else 6.dp
            ),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(if (isCompact) 6.dp else 10.dp)
    ) {
        // Handlebar Canvas
        Canvas(
            modifier = Modifier.size(
                width = if (isCompact) 36.dp else 48.dp,
                height = if (isCompact) 24.dp else 32.dp
            )
        ) {
            val w = size.width
            val h = size.height

            // Base outline
            val baseColor = CardBorder
            val baseStroke = Stroke(width = 3.dp.toPx(), cap = StrokeCap.Round)

            // Outer handlebar path
            val path = Path().apply {
                moveTo(w * 0.1f, h * 0.1f)
                lineTo(w * 0.1f, h * 0.65f)
                lineTo(w * 0.35f, h * 0.85f)
                lineTo(w * 0.65f, h * 0.85f)
                lineTo(w * 0.9f, h * 0.65f)
                lineTo(w * 0.9f, h * 0.1f)
            }
            drawPath(path, color = baseColor, style = baseStroke)

            // Highlight path
            val highlightColor = TopBarGreenDark
            val highlightStroke = Stroke(width = 4.dp.toPx(), cap = StrokeCap.Round)

            when (position) {
                HandPosition.POSITION_1 -> {
                    // Center horizontal crossbar
                    drawLine(
                        color = highlightColor,
                        start = androidx.compose.ui.geometry.Offset(w * 0.35f, h * 0.85f),
                        end = androidx.compose.ui.geometry.Offset(w * 0.65f, h * 0.85f),
                        strokeWidth = highlightStroke.width,
                        cap = StrokeCap.Round
                    )
                }
                HandPosition.POSITION_2 -> {
                    // Left and right corners
                    drawLine(
                        color = highlightColor,
                        start = androidx.compose.ui.geometry.Offset(w * 0.12f, h * 0.55f),
                        end = androidx.compose.ui.geometry.Offset(w * 0.35f, h * 0.85f),
                        strokeWidth = highlightStroke.width,
                        cap = StrokeCap.Round
                    )
                    drawLine(
                        color = highlightColor,
                        start = androidx.compose.ui.geometry.Offset(w * 0.88f, h * 0.55f),
                        end = androidx.compose.ui.geometry.Offset(w * 0.65f, h * 0.85f),
                        strokeWidth = highlightStroke.width,
                        cap = StrokeCap.Round
                    )
                }
                HandPosition.POSITION_3 -> {
                    // Bullhorns
                    drawLine(
                        color = highlightColor,
                        start = androidx.compose.ui.geometry.Offset(w * 0.1f, h * 0.1f),
                        end = androidx.compose.ui.geometry.Offset(w * 0.1f, h * 0.45f),
                        strokeWidth = highlightStroke.width,
                        cap = StrokeCap.Round
                    )
                    drawLine(
                        color = highlightColor,
                        start = androidx.compose.ui.geometry.Offset(w * 0.9f, h * 0.1f),
                        end = androidx.compose.ui.geometry.Offset(w * 0.9f, h * 0.45f),
                        strokeWidth = highlightStroke.width,
                        cap = StrokeCap.Round
                    )
                }
            }
        }

        if (showTitle) {
            Column {
                Row(verticalAlignment = Alignment.Bottom) {
                    Text(
                        text = "把位",
                        fontSize = if (isCompact) 10.sp else 11.sp,
                        fontWeight = FontWeight.Bold,
                        color = TextSecondary
                    )
                    Spacer(modifier = Modifier.width(3.dp))
                    Text(
                        text = "${position.value}",
                        fontSize = if (isCompact) 14.sp else 16.sp,
                        fontWeight = FontWeight.Black,
                        color = TopBarGreenDark
                    )
                }
                Text(
                    text = when (position) {
                        HandPosition.POSITION_1 -> "平把中段"
                        HandPosition.POSITION_2 -> "橫桿轉折"
                        HandPosition.POSITION_3 -> "前端牛角"
                    },
                    fontSize = if (isCompact) 9.sp else 11.sp,
                    fontWeight = FontWeight.Medium,
                    color = TextPrimary
                )
            }
        }
    }
}

File: ios/FitnessRider/Views/Components/HandPositionBadge.swift
import SwiftUI

/// Visual spinning bike handlebar widget highlighting Hand Position 1, 2, or 3
public struct HandPositionBadge: View {
    public let position: HandPosition
    public var isCompact: Bool = false
    public var showTitle: Bool = true

    public init(position: HandPosition, isCompact: Bool = false, showTitle: Bool = true) {
        self.position = position
        self.isCompact = isCompact
        self.showTitle = showTitle
    }

    public var body: some View {
        HStack(spacing: isCompact ? 6 : 10) {
            // Handlebar Schematic Vector
            HandlebarShape(activePosition: position)
                .stroke(FitnessRiderTheme.cardBorder, lineWidth: 3)
                .background(
                    HandlebarHighlight(activePosition: position)
                        .stroke(FitnessRiderTheme.topBarGreenDark, style: StrokeStyle(lineWidth: 4, lineCap: .round))
                )
                .frame(width: isCompact ? 36 : 48, height: isCompact ? 24 : 32)

            if showTitle {
                VStack(alignment: .leading, spacing: 1) {
                    HStack(spacing: 4) {
                        Text("把位")
                            .font(.system(size: isCompact ? 10 : 11, weight: .bold))
                            .foregroundColor(FitnessRiderTheme.textSecondary)
                        Text("\(position.rawValue)")
                            .font(.system(size: isCompact ? 14 : 16, weight: .black, design: .rounded))
                            .foregroundColor(FitnessRiderTheme.topBarGreenDark)
                    }

                    Text(positionLabel)
                        .font(.system(size: isCompact ? 9 : 11, weight: .medium))
                        .foregroundColor(FitnessRiderTheme.textPrimary)
                }
            }
        }
        .padding(.horizontal, isCompact ? 8 : 12)
        .padding(.vertical, isCompact ? 4 : 6)
        .background(Color.white)
        .cornerRadius(8)
        .overlay(
            RoundedRectangle(cornerRadius: 8)
                .stroke(FitnessRiderTheme.cardBorder, lineWidth: 1)
        )
    }

    private var positionLabel: String {
        switch position {
        case .position1: return "平把中段"
        case .position2: return "橫桿轉折"
        case .position3: return "前端牛角"
        }
    }
}

// MARK: - Handlebar Schematic Path
private struct HandlebarShape: Shape {
    let activePosition: HandPosition

    func path(in rect: CGRect) -> Path {
        var path = Path()
        let w = rect.width
        let h = rect.height

        // Outer bullhorns + crossbar geometry of indoor cycle
        // Left horn
        path.move(to: CGPoint(x: w * 0.1, y: h * 0.1))
        path.addLine(to: CGPoint(x: w * 0.1, y: h * 0.65))
        // Left bend to crossbar
        path.addLine(to: CGPoint(x: w * 0.35, y: h * 0.85))
        // Center crossbar
        path.addLine(to: CGPoint(x: w * 0.65, y: h * 0.85))
        // Right bend
        path.addLine(to: CGPoint(x: w * 0.9, y: h * 0.65))
        // Right horn
        path.addLine(to: CGPoint(x: w * 0.9, y: h * 0.1))

        return path
    }
}

private struct HandlebarHighlight: Shape {
    let activePosition: HandPosition

    func path(in rect: CGRect) -> Path {
        var path = Path()
        let w = rect.width
        let h = rect.height

        switch activePosition {
        case .position1:
            // Center flat crossbar (Position 1)
            path.move(to: CGPoint(x: w * 0.35, y: h * 0.85))
            path.addLine(to: CGPoint(x: w * 0.65, y: h * 0.85))
        case .position2:
            // Corners / transitions (Position 2)
            path.move(to: CGPoint(x: w * 0.12, y: h * 0.55))
            path.addLine(to: CGPoint(x: w * 0.35, y: h * 0.85))
            path.move(to: CGPoint(x: w * 0.88, y: h * 0.55))
            path.addLine(to: CGPoint(x: w * 0.65, y: h * 0.85))
        case .position3:
            // Upper bullhorns (Position 3)
            path.move(to: CGPoint(x: w * 0.1, y: h * 0.1))
            path.addLine(to: CGPoint(x: w * 0.1, y: h * 0.45))
            path.move(to: CGPoint(x: w * 0.9, y: h * 0.1))
            path.addLine(to: CGPoint(x: w * 0.9, y: h * 0.45))
        }

        return path
    }
}
